src/Electrical/Digital/logic.jl

```diff
@@ -0,0 +1,68 @@
+struct Logic
+    level::Int
+end
+
+function Base.show(io::IO, ::MIME"text/plain", l::Logic)
+    if l.level == 1
+        print(io, "U")
+    elseif l.level == 2
+        print(io, "X")
+    elseif l.level == 3
+        print(io, "F0")
+    elseif l.level == 4
+        print(io, "F1")
+    elseif l.level == 5
+        print(io, "Z")
+    elseif l.level == 6
+        print(io, "W")
+    elseif l.level == 7
+        print(io, "L")
+    elseif l.level == 8
+        print(io, "H")
+    elseif l.level == 9
+        print(io, "DC")
+    else
+        print(io, "Invalid logic level")
+    end
+end
+
+# Aliases for 9 Digital Logic levels
+const Uninitialized = Logic(1)
+const ForcingUnknown = Logic(2)
+const ForcingZero = Logic(3)
+const ForcingOne = Logic(4)
+const HighImpedence = Logic(5)
+const WeakUnknown = Logic(6)
+const WeakZero = Logic(7)
+const WeakOne = Logic(8)
+const DontCare = Logic(9)
```

Comment:
This feels an awful lot like an `@enum`. Have you considered using an enumeration for this type instead?

Reply:
Initially I was considering to include 1 and 0 as is in the set of logic levels; as @enum wouldn't work with that, I dropped the idea then.

Now because of F0, F1 it is doable - Logic as an Enum of these 9 vals.
ATM I can't think of any other field Logic could have. So I will swap this part with an Enum

Reply:
`getfield.(<big StructLogic matrix>, :levels)` vs `Int.(<big EnumLogic matrix>)`
Speeds were very close, and the former was slightly faster. However, the latter always allocated less.